gnomad/variant_qc/vqsr.py

```diff
@@ -0,0 +1,988 @@
+            disk_size=small_disk,
+        ).out_tranches
+
+        # 3. Apply recalibration
+        # Doesn't require too much storage in scatter mode (<500MB on gnomad VCF for each scatter), use small_disk
+        scattered_vcfs = [
+            apply_recalibration(
+                b=b,
+                input_vcf=sites_only_vcf,
+                out_vcf_name=output_vcf_name,
+                indels_recalibration=indels_recalibrations[idx],
+                indels_tranches=indels_gathered_tranches,
+                snps_recalibration=snps_recalibrations[idx],
+                snps_tranches=snps_gathered_tranches,
+                utils=utils,
+                disk_size=small_disk,
+                use_as_annotations=use_as_annotations,
+                scatter=idx,
+                interval=intervals[f'interval_{idx}'],
+                out_bucket=out_bucket,
+            ).output_vcf
+            for idx in range(utils['NUMBER_OF_GENOMICS_DB_INTERVALS'])
+        ]
+
+        # 4. Gather VCFs
+        recalibrated_gathered_vcf_job = gather_vcfs(
+            b=b,
+            input_vcfs=scattered_vcfs,
+            out_vcf_name=output_vcf_name,
+            utils=utils,
+            disk_size=huge_disk,
+            out_bucket=out_bucket,
+        )
+
+    else:
+        snps_recalibrator_job = snps_variant_recalibrator(
+            b=b,
+            sites_only_vcf=sites_only_vcf,
+            utils=utils,
+            out_bucket=out_bucket,
+            use_as_annotations=use_as_annotations,
+            transmitted_singletons_resource_vcf=transmitted_singletons,
+            sibling_singletons_resource_vcf=sibling_singletons,
+            max_gaussians=snp_max_gaussians,
+        )
+        snps_recalibration = snps_recalibrator_job.recalibration
+        snps_tranches = snps_recalibrator_job.tranches
+
+        indels_variant_recalibrator_job = indels_variant_recalibrator(
+            b=b,
+            sites_only_vcf=sites_only_vcf,
+            transmitted_singletons_resource_vcf=transmitted_singletons,
+            sibling_singletons_resource_vcf=sibling_singletons,
+            utils=utils,
+            out_bucket=out_bucket,
+            use_as_annotations=use_as_annotations,
+            max_gaussians=indel_max_gaussians,
+        )
+        indels_recalibration = indels_variant_recalibrator_job.recalibration
+        indels_tranches = indels_variant_recalibrator_job.tranches
+
+        recalibrated_gathered_vcf_job = apply_recalibration(
+            b=b,
+            input_vcf=sites_only_vcf,
+            out_vcf_name=output_vcf_name,
+            indels_recalibration=indels_recalibration,
+            indels_tranches=indels_tranches,
+            snps_recalibration=snps_recalibration,
+            snps_tranches=snps_tranches,
+            utils=utils,
+            disk_size=huge_disk,
+            use_as_annotations=use_as_annotations,
+            out_bucket=out_bucket,
+        )
+
+    # return recalibrated_gathered_vcf_job
```

Comment:
does this function not return anything?

Reply:
Oh I see it now, gather_vcfs() and apply_recalibration() would write to the bucket that you want in the end.